openvdb/Makefile

```diff
@@ -282,6 +284,41 @@ ifneq (2,$(strip $(GLFW_MAJOR_VERSION)))
     CXXFLAGS += -DOPENVDB_USE_GLFW_3
 endif
 
+ifeq (yes,$(strip $(warnings)))
+    USING_CLANG=$(shell ${CXX} -dM -E -x c++ - < /dev/null | grep __clang__)
+    USING_GCC=$(shell ${CXX} -dM -E -x c++ - < /dev/null | grep __GNUC__)
```

Comment:
The most reliable way I've been able to come up with for detecting gcc vs clang irrespective of any wrappers around CXX, assuming that either gcc or clang has indeed been called. clang also defines __GNUC__.

Reply:
ICC also defines `__GNUC__`, but its diagnostic flags are incompatible.

How about "`${CXX} --version | grep clang`", "`${CXX} --version | grep GCC`", etc.?

Reply:
Yeah I considered this but for some reason thought querying the defines would be more 'correct'. Not sure why, I'll change this